app/core/ingestion/validator.py

```diff
@@ -55,10 +59,59 @@ def validate_document_row(
     if result.type != ResultType.OK:
         errors.append(result)
 
+    # Check metadata
     result, _ = build_metadata(taxonomy, row)
     if result.type != ResultType.OK:
         errors.append(result)
 
+    on_row = f"on row {row.row_number}"
+    # Check family
+    family_id = row.cpr_family_id
+
+    if family_id in context.mde.families.keys():
```

Comment:
Not only is this call to `keys()` redundant, it turns an O(1) lookup into an O(n) lookup... :grin: